packages/adapter-prisma/lib/adapter-prisma.js

```diff
@@ -0,0 +1,611 @@
+                } else if (r.cardinality === '1:1') {
+                  return [`${f.path} ${f.refListKey}? @relation("${relName}")`];
+                } else {
+                  return [`${f.path} ${f.refListKey}[] @relation("${relName}")`];
+                }
+              }
+            })
+        ),
+        ...flatten(
+          rels
+            .filter(({ right }) => !right)
+            .filter(({ left }) => left.refListKey === listAdapter.key)
+            .filter(({ cardinality }) => cardinality === 'N:N')
+            .map(({ left: { path, listKey }, tableName }) => [
+              `from_${path} ${listKey}[] @relation("${tableName}", references: [id])`,
+            ])
+        ),
+      ];
+
+      return `
+        model ${listAdapter.key} {
+          id Int @id @default(autoincrement())
+          ${[...scalarFields, ...relFields].join('\n  ')}
+        }`;
+    });
+
+    const enums = flatten(
+      Object.values(this.listAdapters).map(listAdapter =>
+        flatten(
+          listAdapter.fieldAdapters
+            .filter(f => !f.field.isRelationship)
+            .filter(f => f.path !== 'id')
+            .map(f => f.getPrismaEnums())
+        )
+      )
+    );
+
+    const header = `
+      datasource postgresql {
+        url      = env("DATABASE_URL")
+        provider = "postgresql"
+      }
+      generator client {
+        provider = "prisma-client-js"
+        output = "${clientDir}"
+        previewFeatures = ["insensitiveFilters"]
+      }`;
+    return await formatSchema({ schema: header + models.join('\n') + '\n' + enums.join('\n') });
+  }
+
+  async postConnect({ rels }) {
+    Object.values(this.listAdapters).forEach(listAdapter => {
+      listAdapter._postConnect({ rels, prisma: this.prisma });
+    });
+
+    if (this.config.dropDatabase && process.env.NODE_ENV !== 'production') {
+      await this.dropDatabase();
+    }
+    return [];
+  }
+
+  // This will drop all the tables in the backing database. Use wisely.
+  async dropDatabase() {
+    let migrationNeeded = true;
+    for (const { tablename } of await this.prisma.$queryRaw(
+      `SELECT tablename FROM pg_tables WHERE schemaname='${this.dbSchemaName}'`
+    )) {
+      if (tablename.includes('_Migration')) {
+        migrationNeeded = false;
+      }
+      await this.prisma.$queryRaw(
+        `TRUNCATE TABLE \"${this.dbSchemaName}\".\"${tablename}\" CASCADe;`
+      );
+    }
+    if (migrationNeeded) {
+      this._saveMigration({ name: 'init' });
+      this._executeMigrations();
+    }
+  }
+
+  _saveMigration({ name }) {
+    this._runPrismaCmd(`migrate save --name ${name} --experimental`);
+  }
+
+  _executeMigrations() {
+    this._runPrismaCmd('migrate up --experimental');
+  }
+
+  _runPrismaCmd(cmd) {
+    return execSync(`yarn prisma ${cmd} --schema ${this.schemaPath}`, {
+      env: { ...process.env, DATABASE_URL: this._url() },
+      encoding: 'utf-8',
+    });
+  }
+
+  disconnect() {
+    this.prisma.$disconnect();
```

Comment:
I assume this returns a promise so we should return it?

Reply:
Yep, good call 👍